onnxruntime/core/providers/cuda/nn/dropout.cc

```diff
@@ -3,50 +3,83 @@
 
 #include "core/providers/cuda/nn/dropout.h"
 
+#include "core/providers/cuda/nn/dropout_impl.h"
+
 namespace onnxruntime {
 namespace cuda {
 
-ONNX_OPERATOR_VERSIONED_KERNEL_EX(
-    Dropout,
-    kOnnxDomain,
-    12, 12,
-    kCudaExecutionProvider,
-    (*KernelDefBuilder::Create())
-        .TypeConstraint("T", DataTypeImpl::AllIEEEFloatTensorTypes())
-        .TypeConstraint("T1", DataTypeImpl::AllIEEEFloatTensorTypes())
-        .TypeConstraint("T2", DataTypeImpl::GetTensorType<bool>())
-        .InputMemoryType(OrtMemTypeCPUInput, 1)
-        .InputMemoryType(OrtMemTypeCPUInput, 2),
-    Dropout);
-
-ONNX_OPERATOR_KERNEL_EX(
-    Dropout,
-    kOnnxDomain,
-    13,
-    kCudaExecutionProvider,
-    (*KernelDefBuilder::Create())
-        .TypeConstraint("T", BuildKernelDefConstraints<MLFloat16, float, double, BFloat16>())
-        .TypeConstraint("T1", BuildKernelDefConstraints<MLFloat16, float, double, BFloat16>())
-        .TypeConstraint("T2", DataTypeImpl::GetTensorType<bool>())
-        .InputMemoryType(OrtMemTypeCPUInput, 1)
-        .InputMemoryType(OrtMemTypeCPUInput, 2),
-    Dropout);
-
-Status Dropout::ComputeInternal(OpKernelContext* context) const {
-  //Get X_data
+namespace {
+
+// Bitmask tensor is uint_32 type.
+using BitmaskElementType = uint32_t;
+constexpr int64_t kNumBitsPerBitmaskElement = static_cast<int64_t>(std::numeric_limits<BitmaskElementType>::digits);
+
+template <typename T>
+struct GetRatioDataImpl {
+  void operator()(const Tensor* ratio, float& ratio_data) const {
+    ratio_data = static_cast<float>(*(ratio->template Data<T>()));
+    ORT_ENFORCE(ratio_data >= 0.0f && ratio_data < 1.0f, "ratio_data is outside range [0, 1)");
+  }
+};
+
+template <typename T>
+struct DropoutComputeImpl {
+  void operator()(const cudaDeviceProp& prop, cudaStream_t stream, const int64_t N, const int64_t mask_element_count,
+                  const float ratio_data, PhiloxGenerator& generator, const Tensor& X, Tensor& Y, void* mask_data,
+                  bool use_bitmask) const {
+    typedef typename ToCudaType<T>::MappedType CudaT;
+    const CudaT* X_data = reinterpret_cast<const CudaT*>(X.template Data<T>());
+    CudaT* Y_data = reinterpret_cast<CudaT*>(Y.template MutableData<T>());
+
+    DropoutKernelImpl<CudaT>(prop, stream, N, mask_element_count, ratio_data, generator, X_data, Y_data, mask_data,
+                             use_bitmask);
+  }
+};
+
+}  // namespace
+
+ONNX_OPERATOR_VERSIONED_KERNEL_EX(Dropout, kOnnxDomain, 12, 12, kCudaExecutionProvider,
+                                  (*KernelDefBuilder::Create())
```

Comment:
can we avoid those unnecessary format changes?

Reply:
Sometimes most part of the file is changed and the changed lines are not contiguous, it's easy to just "format the document" instead of selecting the changed parts piece by piece and "format the selection".